### DISCUSSION TITLE
how to call resize in sample client?

### DISCUSSION THREAD

**Discussion by kimcoldwater** [FreeRDP/FreeRDP#8108] [Q&A] [answered]
hello,

I customize FreeRDP and make it available on the web.

I Successed connection and event delivery and clipboard by customizing the sample client (headless) on Linux.

However, resize cannot be easily resolved.

I'm making it using java-jni.

I delivered `gdi_resize` like resize of existing sample client, but only the screen size has changed, but the resolution has not changed.

```
    JNIEXPORT void jni_send_resize(JNIEnv* env, jclass cls, jlong instance, jint resizeWidth , jint resizeHeight){

        freerdp* inst = (freerdp*)instance;
        rdpContext* context = (rdpContext*)inst->context;
    	rdpGdi* gdi;
    	rdpSettings* settings;
    	settings = context->settings;

    	settings->DesktopWidth = resizeWidth;
        settings->DesktopHeight = resizeHeight;
    	gdi = context->gdi;

        gdi_resize(gdi, settings->DesktopWidth, settings->DesktopHeight);
    }
```

It's the same call method as the Android version.

But I already know this is the wrong way.

Is there a way to call `resize` in internally?

Is there a function that is called when you increase or decrease the screen with the mouse in the existing X11 version?

Examples of other clients were not headless versions and I could not be referenced.

Could you advise me on how to call `resize` from the headless sample client?

**Comment by akallabeth** [comment, marked as answer]
* In RDP you can only request the session resolution at connection time (`DesktopWidth`, `DesktopHeight`)
* The server may inform the client that the resolution changed (that is where the `DesktopResize` comes to play), to make matters worse `gfx` has its own callbacks for that
* If you want the client to change resolution in a running connection, you need the `disp` channel (that informs the server about the clients preference and may act accordingly). That channel is not without its own issues though, if you request too fast the server session might crash...

**Reply by kimcoldwater** [reply]
Let's try to configure the disp channel. Thank you.
